source/dofs/dof_tools_constraints.cc

```diff
@@ -1599,10 +1619,29 @@ namespace DoFTools
                             fes.insert(neighbor_fe_index);
                             const dealii::hp::FECollection<dim, spacedim>
                               &fe_collection = dof_handler.get_fe_collection();
-                            const unsigned int dominating_fe_index =
-                              fe_collection
-                                .find_least_dominating_fe_in_collection(
-                                  fes, /*codim=*/1);
+
+                            // We first try to find the dominating finite
+                            // element among both elements. If we do not
+                            // succeed, we extend our search on the whole
+                            // collection by picking the least dominating one in
```

Comment:
and here. Since I'm seeing this code block for the fourth time now, think about whether it could be put into a helper function somewhere :-)